multinet/api/tasks/process/d3_json.py

```diff
@@ -0,0 +1,81 @@
+import json
+from typing import BinaryIO, Dict
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+
+from multinet.api.models import Network, Table, Upload
+
+from .utils import complete_upload, fail_upload_with_message
+
+logger = get_task_logger(__name__)
+
+
+def d3_node_to_arango_doc(node: Dict) -> Dict:
+    new_node = dict(node)
+    new_node['_key'] = str(new_node.pop('id'))
+
+    return new_node
+
+
+def d3_link_to_arango_doc(link: Dict, node_table_name: str) -> Dict:
+    new_link = dict(link)
+    new_link['_from'] = f'{node_table_name}/{new_link.pop("source")}'
+    new_link['_to'] = f'{node_table_name}/{new_link.pop("target")}'
+
+    return new_link
+
+
+@shared_task
+def process_d3_json(
+    upload_id: int,
+    network_name: str,
+    node_table_name: str,
+    edge_table_name: str,
+) -> None:
+    logger.info(f'Begin processing of upload {upload_id}')
+    upload: Upload = Upload.objects.get(id=upload_id)
+
+    # Update status
+    upload.status = Upload.UploadStatus.STARTED
+    upload.save()
+
+    # Download data from S3/MinIO
+    with upload.blob as blob_file:
+        blob_file: BinaryIO = blob_file
```

Comment:
It's for typing, as otherwise the type is not inferred correctly.

Reply:
Apparently you can skip the assignment part: https://stackoverflow.com/a/60170752/1886928.

If that works, I'd like to update this line of code so that future practice can take this as a model.

Reply:
Good catch. I think this is what I was thinking of, and for some reason imagined that the assignment was necessary. I'll submit a fix.